### ISSUE TITLE
Issues reproducing RoBERTa results on DocRED

### ISSUE THREAD

**Issue by xieyvhuan** [yanyi74/GREP#2]
Hello, I’m currently trying to reproduce the results of the RoBERTa model on the DocRED dataset, but I encountered some issues:

1. Unexpectedly high dev F1 before ISCF inference

When I run:

bash scripts/run_roberta.bash roberta-1 0.03 70

I obtain an F1 score as high as 66+ on the dev set (before applying ISCF inference). However, this seems to be due to overfitting, since the score on the test set drops to around 63.

Could you kindly share your training scripts, or provide more detailed settings, so I can compare?

2. Model crash when LAMBDA=0

Running:

bash scripts/run_roberta.bash roberta-1 0.03 70

works without issues. But when I run:

bash scripts/run_roberta.bash roberta-1 0 70

the model crashes starting from epoch 7, with the output becoming:

[-100, -100, -100]

I then set LAMBDA=0.0000001 to approximate training without evidence supervision, in order to continue with ISCF inference.

Do you have any insights into the possible reasons for this crash?

Would you suggest a better way to train without evidence supervision?

3. Overfitting issue with dimension fix

In order to resolve dimension mismatch problems, I modified the model as follows:

self.head_extractor2 = nn.Linear(emb_size + self.hidden_size * 2, emb_size)
self.tail_extractor2 = nn.Linear(emb_size + self.hidden_size * 2, emb_size)

However, the model still suffers from severe overfitting.

In this case, would you recommend choosing a checkpoint that performs less optimally on the dev set, but may generalize better?

Thank you very much in advance for your help! 🙏
